### DISCUSSION TITLE
Extracting invidious API client as separate pub.dev (dart) package

### DISCUSSION THREAD

**Discussion by KRTirtho** [lamarios/clipious#616] [Ideas]
Hey, love the project and I kind of use it nowadays in place of youtube. Works really well.

I'm also the author/maintainer of [Spotube](https://spotube.krtirtho.dev/) and in Spotube, we use Piped as a source. 
Piped and Invidious is similar in their aspects with a different API signature. I already published [`piped_client`](https://pub.dev/packages/piped_client) package covering some of the APIs (the ones we need tbh)

We were planning to add invidious as a source too and was about to write it from ground up. Until I found Clipious (never looking back to YouTube hehe).
So we want to use the invidious API client that is used. It'll be awesome if you or us published that part as a separate pub package so the community could benifit from it

What do you say?

**Comment by KRTirtho** [comment]
Upstream discussion https://github.com/KRTirtho/spotube/issues/1948

**Comment by lamarios** [comment]
Hey !

I don't think i'll be able to do this anytime soon for a couple of reason:
- the code is quite bad in some ways, the project went through few major refactors as I was a flutter beginner when i started this project and the service class that talks to invidious is fairly dependent of the local database. I'd need quite a lot of effort to extract this as a dart package
- I don't think I have the bandwidth to maintain an extra thing especially if some other project depends on.

**Reply by s-b-repo** [reply]
> Hey !
> 
> I don't think i'll be able to do this anytime soon for a couple of reason:
> 
>     * the code is quite bad in some ways, the project went through few major refactors as I was a flutter beginner when i started this project and the service class that talks to invidious is fairly dependent of the local database. I'd need quite a lot of effort to extract this as a dart package
> 
>     * I don't think I have the bandwidth to maintain an extra thing especially if some other project depends on.
yeah i understand i think i have a solution why don't you make a share repo for the api it self then we can make you owner and make the main spotube owner a contributer if you could make the base package then the spotube maintains it and improves it

**Reply by s-b-repo** [reply]
*or maintainer whatever you want to call it we can also ask invidious repo maintainers for help or communtiy

**Reply by KRTirtho** [reply]
@lamarios totally understandable. It is hard to maintain multiple projects. Unfortunately, I know this very well.

No worries. I can extract the API client from here (shouldn't take much time) and make a package out of it. Then, if u want u can also depend on it, if it fits your needs and contribute in your time if possible.

@s-b-repo's proposal is also nice. We can rich out to invidious maintainers to help us out occasionally.

**Reply by KRTirtho** [reply]
This would've been matter of minutes if invidious had a openapi schema file 😄
